pkg/cue/process/handle.go

```diff
@@ -235,21 +253,29 @@ func (ctx *templateContext) BaseContextFile() string {
 	}
 
 	if ctx.data != nil {
-		d, _ := json.Marshal(ctx.data)
+		d, err := json.Marshal(ctx.data)
+		if err != nil {
+			return "", err
+		}
 		buff += fmt.Sprintf("\n %s", structMarshal(string(d)))
 	}
 
-	return fmt.Sprintf("context: %s", structMarshal(buff))
+	return fmt.Sprintf("context: %s", structMarshal(buff)), nil
 }
 
 // ExtendedContextFile return cue format string of templateContext and extended secret context
 func (ctx *templateContext) ExtendedContextFile() string {
```

Comment:
we should also return an error here if so

Reply:
If I do so, there will be other refactors required like in `template.go, handle_test.go, task.go`. Let me know if I still need to return an error here.

Reply:
then, let's revert this part of the chage, I think it's fine to omit the error here.

Reply:
hmm! do you mean reverting the changes on `BaseContextFile()` method? 😕

Reply:
yes

Reply:
I'm afrait we will cause confusion if we return error info as the CUE context string back. Actually we must return a clear error type if we define err here.

Reply:
>  Actually we must return a clear error type if we define err here.

Sorry Jianbo! I couldn't get that. Also let me tell you as @FogDong suggested me returning `error` value for BaseContextFile() method.

Reply:
Sorry, let me discuss with Fog.

Reply:
This PR will not return error info as the CUE context string, instead it returns err as an error.

Reply:
Sorry again! I'm still confused. Let me know if I'm wrong here;
```go
func (ctx *templateContext) BaseContextFile() (string, error) {
```
The above method is called by `context, err := ctx.BaseContextFile()` Then if error is not nil, then it returns `return fmt.Sprintf("failed to convert data to application with marshal err %v", err)`.

So, what's the logic of returning error info as the CUE context string? Thanks

Reply:
I think I got that. this `fmt.Sprintf("failed to convert data to application with marshal err %v", err)` should be  in the format of CUE ctx string.

Reply:
If there's err, the string should be `""`.
In your example, if there's err in `context, err := ctx.BaseContextFile()`, it should return `"", err`